src/utils/serve_content.rs

```diff
@@ -0,0 +1,98 @@
+//! Content serving utilities with support for conditional requests
+//! as defined in [RFC 7232](https://tools.ietf.org/html/rfc7232)
+
+use async_std::io::{BufRead as AsyncBufRead, Read as AsyncRead, Seek as AsyncSeek};
+
+use crate::{Request, Response, Result, StatusCode};
+
+/// A HTTP ressource modification state.
```

Comment:
```suggestion
/// A HTTP resource modification state.
```

And other uses of the word